File: contains_duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # HashTable O(n) complexity
        mp = {
            # num_in_list : # count
        }
        # Runtime: O(n)
        for i in nums:
            if i not in mp.keys():
                mp[i] = 1
            else:
                return True

        return False

File: contains_duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> hs = new HashSet<>();

        for (var i : nums) {
            if (hs.contains(i)) {
                return true;
            } else {
                hs.add(i);
            }
        }
        return false;
    }
}
